modules/monitoring-configuring-shiftstack-scraping.adoc

```diff
@@ -0,0 +1,90 @@
+// Module included in the following assemblies:
+//
+// * observability/monitoring/shiftstack-prometheus-configuration.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="monitoring-configuring-shiftstack-scraping_{context}"]
+= Collecting cluster metrics from the federation endpoint
+
+// TODO: Is this RHOSO only?
+
+You can open the federation endpoint of a cluster to make metrics available to a Red Hat OpenStack Services on OpenShift (RHOSO) cluster to practice pull-based monitoring.
+
+.Prerequisites
+
+- You have access to an external Prometheus instance.
+- You have administrative access to {rh-openstack} and your cluster.
+- You have certificates for secure communication with mTLS.
+- Telemetry is enabled in the {rh-openstack} environment.
+- The Cluster Observability Operator is installed on your cluster.
+- The monitoring stack for your cluster is configured.
+- Your cluster has its federation endpoint exposed.
+// That is, https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/monitoring/accessing-third-party-monitoring-apis#monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_accessing-monitoring-apis-by-using-the-cli
+
+.Procedure
+
+. Connect to your cluster by using a username and password; do not log in by using a `kubeconfig` file that was generated by the installation program.
+
+. To retrieve a token from the cluster, run the following command:
++
+[source,terminal]
+----
+$ oc whoami -t
+----
+
+. To get the Prometheus federation route URL from your cluster, enter the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring get route prometheus-k8s-federate -ojsonpath={'.status.ingress[].host'}
+----
+
+. Write a manifest for a scrape configuration and save it as a file called `cluster-scrape-config.yaml`. As an example:
++
+[source,yaml]
+----
+apiVersion: monitoring.rhobs/v1alpha1
+kind: ScrapeConfig
+metadata:
+  labels:
+    service: metricStorage
+  name: sos1-federated
+  namespace: openstack
+spec:
+  params:
+    'match[]':
+    - '{__name__=~"kube_node_info|kube_persistentvolume_info"}' # <1>
+  metricsPath: '/federate'
+  authorization:
+    type: Bearer
+    credentials:
+      name: ocp-federated # <2>
+      key: token
+  scheme: HTTPS # or HTTP
+  scrapeInterval: 30s # <3>
+  staticConfigs:
+  - targets:
+    - prometheus-k8s-federate-openshift-monitoring.apps.openshift.example # <4>
+----
+<1> Add metrics here. In this example, only the metrics `kube_node_info` and `kube_persistentvolume_info` are requested.
+<2> Note the credential name for later.
+<3> Limit scraping to fewer than 1000 samples for each request with a maximum frequency of once every 30 seconds.
+<4> Insert the URL you fetched previously here. If the endpoint is HTTPS and uses a custom certificate authority, add a `tlsConfig` section after it.
+
+. While connected to the RHOSO cluster, apply the manifest by running the following command:
```

Comment:
It's "while connected to the OpenShift cluster hosting the RHOSO control plane" actually, but that might be a bit too long. Not sure that everyone understand that by "RHOSO" cluster, we're in fact talking about the OpenShift cluster it's running on.

Reply:
I'll take verbosity over vagueness. 

```suggestion
. While connected to the {product-title} cluster that hosts the RHOSO control plane, apply the manifest by running the following command:
```

Reply:
@pierreprinetti You're correct! 

We don't need to use `{product-title}` at all if it complicates things. In a dumb substitution sense, it applies to OpenShift core in most [permutations](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#product-name-and-version) of the documentation. Most writers, self included, unthinkingly sub in `{product-title}` whenever we see `OpenShift` or things like it.

I need to consult with my CS, I think, as there are branding issues here as well as ones of elegance in writing. We'll probably end up with a new attribute specifically for this situation.

Reply:
@pierreprinetti Any thoughts on "management cluster" vs. "workload cluster"?

Reply:
RHOSO users may have internalised the idea that "management cluster" is the name of the OpenShift cluster hosting RHOSO's control plane.

I find "workload cluster" trickier, because I imagine "workload" as being "the actual software that runs your business logic, as opposed to the infrastructure stack"... but the OpenShift cluster that runs on top of OpenStack is not your business logic application, it's just the last infrastructure layer under it.

That being said: if you find a better name, rest assured I will use it from now on.